contracts/tests/test_deposit_locker.py

```diff
@@ -213,5 +166,190 @@ def test_event_slash(
         fromBlock=latest_block_number
     ).get_all_entries()[0]["args"]
 
-    assert event["validator"] == malicious_validator_address
+    assert event["slashedDepositor"] == malicious_validator_address
     assert event["slashedValue"] == deposit_amount
+
+
+@attr.s(auto_attribs=True)
+class Env:
+    deposit_locker: Any
+    slasher: Any
+    proxy: Any
+    depositors: List[Any]
+    other_accounts: List[Any]
+
+    def register_depositor(self, depositor):
+        return self.deposit_locker.functions.registerDepositor(depositor).transact(
+            {"from": self.proxy}
+        )
+
+    def register_all_depositors(self):
+        for d in self.depositors:
+            self.register_depositor(d)
+
+    def deposit(self, value_per_depositor, total_value):
+        return self.deposit_locker.functions.deposit(value_per_depositor).transact(
+            {"from": self.proxy, "value": total_value}
+        )
+
+    def slash(self, depositor):
+        return self.deposit_locker.functions.slash(depositor).transact(
+            {"from": self.slasher}
+        )
+
+    def withdraw(self, depositor):
+        return self.deposit_locker.functions.withdraw().transact({"from": depositor})
+
+    def can_withdraw(self, depositor):
+        return self.deposit_locker.functions.canWithdraw(depositor).call()
+
+
+@pytest.fixture()
+def testenv(deploy_contract, accounts, web3):
```

Comment:
I guess this should be a `deposit_locker_testenv` ?

Reply:
no, the fixture lives in a file called test_deposit_locker. That is enough.